gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -1029,6 +1034,17 @@ public GradeSaveResponse saveExcuse(final Long assignmentId, final String studen
 	 */
 	public List<GbUser> getGbUsersForUiSettings(List<String> userUuids, GradebookUiSettings settings, Site site) {
 
+		Map<String, List<String>> userSections = new HashMap<>();
+		for (CourseSection cs : sectionManager.getSections(site.getId())) {
+			for (EnrollmentRecord er : sectionManager.getSectionEnrollments(cs.getUuid())) {
+				String userId = er.getUser().getUserUid();
+				if (!userSections.containsKey(userId)) {
```

Comment:
I think it would be more efficient to just do the `.get()` and check for `null` return, rather than doing both a `containsKey()` and a `.get()` (you're effectively doing two bucket searches, rather than one). For example:

```
List<String> sections = userSections.get(userId);
if (sections == null)
{
    sections = new ArrayList<>();
}
sections.add(cs.getTitle());
userSections.put(userId, sections);
```